test/common/stats/utility_test.cc

```diff
@@ -273,6 +275,29 @@ TEST_P(StatsUtilityTest, ScopeTextReadoutOnce) { scopeOnce<TextReadout>(makeText
 
 TEST_P(StatsUtilityTest, ScopeTextReadoutAll) { scopeAll<TextReadout>(makeTextReadout()); }
 
+TEST_P(StatsUtilityTest, SanitizeStatsName) {
+
+  absl::string_view testStrings[6] = {
+    "a.b.c.",
+    ".a.b.c",
+    "a::b",
+    "a.\0",
+    "a://b",
+    "a:/b"
+  };
+    absl::string_view expectStrings[6] = {
+    "a.b.c",
+    "a.b.c",
+    "a__b",
+    "a",
+    "a_b",
+    "a_b"
+  };
+  for (size_t i = 0; i < std::size(testStrings); i++) {
+    EXPECT_EQ(expectStrings[i], Utility::sanitizeStatsName(testStrings[i]));
```

Comment:
looks great. Up to you, but maybe it'd be more readable to write this with one EXPECT_EQ per line, e.g.
```
 EXPECT_EQ("a.b.c", Utility::sanitizeStatsName("a.b.c."));
 EXPECT_EQ("a.b.c", Utility::anitizeStatsName(".a.b.c."));
 ...
```
it'd be fewer lines, your eye would look at input and output together, and if any of them fails then gtest will print the line number telling you exactly which one.

Reply:
I will do it.

Reply:
Please review again.